accounts/abi/bind/backends/simulated.go

```diff
@@ -645,9 +656,13 @@ func (b *SimulatedBackend) SendTransaction(ctx context.Context, tx *types.Transa
 	// Include tx in chain
 	blocks, _ := core.GenerateChain(b.config, block, ethash.NewFaker(), b.database, 1, func(number int, block *core.BlockGen) {
 		for _, tx := range b.pendingBlock.Transactions() {
+			if b.marketGasPrice == nil || b.marketGasPrice.Cmp(tx.GasPrice())  <= 0 {
+				block.AddTxWithChain(b.blockchain, tx)
+			}
+		}
+		if b.marketGasPrice == nil || b.marketGasPrice.Cmp(tx.GasPrice())  <= 0 {
 			block.AddTxWithChain(b.blockchain, tx)
 		}
```

Comment:
could use a small comment explaining what is being checked here
also, there could be scenarios where only some of the pending transactions and sent transactions get added to the block. Will this zero out the remaining "low gas price" pending transactions?

Reply:
so i tried to only have changes in the `Commit()` function and keep the `SendTransaction` intact and it seems like if we dont modify the `pendingBlock` we are unable to do duplicate nonce in simulated chain.

anything that is in our pendingBlock even when its not in the chain when we `commit()` it changes the state of the db, any new transaction check this state of the db and requires the nonce has to be greater than whatever is in the pendingBlock.  

i did add a stuck transactions field and have test against that scenario, let me know if that works. Thank you!